tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/RawTensor.java

```diff
@@ -219,9 +221,16 @@ private static int dtype(TF_Tensor handle) {
   RawTensor(TensorTypeInfo<? extends TType> typeInfo, Shape shape) {
     this.typeInfo = typeInfo;
     this.shape = shape;
+
+    TensorScope currentScope = TensorScope.currentScope();
+    if (currentScope != null) {
+      this.scope = currentScope.withAttached(this);
+    }
   }
 
   private PointerScope tensorScope;
+  private boolean closed;
+  TensorScope scope;
```

Comment:
Would be clearer if we revert the naming here, i.e. `scope` should become `tensorScope`, and the actual `tensorScope` should `pointerScope` or `nativeScope`, something like that.